lib/client.js

```diff
@@ -266,3 +275,46 @@ function _processHTTPError(error) {
   }
   return {cause, httpStatusCode};
 }
+
+async function _signRequest({url, json, signer}) {
```

Comment:
Thinking about this more, we should move `_signRequest` and `_verifyRequest` into the signature.js helper -- and make them use a common array for the headers so that they don't get de-synced. It also seems like the right place to put the crypto details vs. having them spread across client/server.